app/controllers/v0/terms_of_use_agreements_controller.rb

```diff
@@ -8,7 +8,12 @@ class TermsOfUseAgreementsController < ApplicationController
 
     skip_before_action :verify_authenticity_token, only: [:update_provisioning]
     skip_before_action :authenticate
-    before_action :terms_authenticate
+    before_action :terms_authenticate, except: [:current_status]
+
+    def current_status
+      agreement_status = find_terms_of_use_agreement_by_icn(params[:icn])
+      render_success(action: 'current_status', body: { agreement_status: }, icn: params[:icn])
```

Comment:
❓ So, we are OK with an HTTP 200 for requests with an ICN that doesn't have an associated terms of use agreement? I think that's fine from a log parsing perspective, but would a different status make more sense? I think a 404 would be confusing (is it the URL or resource that isn't found?), but maybe a 204 no content would work better.

Reply:
I agree that we should return a 404 if an agreement is not found

Reply:
So the reason for this behavior is to give as little information as possible since it's unauthenticated. If it returned 404 an attacker could use this to sweep all possible ICN and see which exist and don't. This still gives some info about whether a user has recently used an ICN but I think this is better than giving the ability to potentially know all existing ICN

Reply:
If you all can think of a way to give even less info that would be good too. But MHV needs to specifically know: given an ICN, has it accepted latest terms, has it denied latest terms, or neither

Reply:
It's possible to have UserAccounts that don't have connected TermsOfUseAgreements though. Would this be a place for 204 - No Content? It is technically returning an object with `agreement_status` as `null`.

Otherwise I would agree that 404 makes sense.